src/mysqlsh_helpers.py

```diff
@@ -56,19 +73,45 @@ def configure_mysql_users(self):
         """
         pass
 
-    def configure_instance(self):
+    def configure_instance(self) -> None:
         """Configure the instance to be used in an InnoDB cluster."""
-        pass
+        _script = f"""
+dba.configure_instance('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')
+my_shell = shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')
+my_shell.run_sql("RESTART;");
+        """
 
-    def create_cluster(self):
+        try:
+            logger.debug("Configuring instance for InnoDB")
+            self._run_mysqlsh_script(_script)
+
+            logger.debug("Waiting until MySQL is restarted")
+            self._wait_until_mysql_connection()
+        except subprocess.CalledProcessError as e:
+            logger.exception(f"Failed to configure instance: {self.instance_address}", exc_info=e)
+            raise MySQLInstanceConfigureError(e.stderr)
+
+    def create_cluster(self) -> None:
         """Create an InnoDB cluster with Group Replication enabled."""
         pass
 
-    def add_instance_to_cluster(self):
+    def add_instance_to_cluster(self) -> None:
         """Add an instance to the InnoDB cluster."""
         pass
 
-    def run_mysqlsh_script(self, script: str) -> AnyStr:
+    @retry(reraise=True, stop=stop_after_delay(30), wait=wait_fixed(5))
+    def _wait_until_mysql_connection(self) -> None:
+        """Wait until a connection to MySQL has been obtained.
+
+        Retry every 5 seconds for 30 seconds if there is an issue obtaining a connection.
+        """
+        _script = f"""
+my_shell = shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')
```

Comment:
It's not possible to set a timeout to `shell.connect` ?

Reply:
there's a default connect timeout of 10s. if the server is unavailable, i believe that the command will fail right away - and result in retries every 5s for 30s. however, if the server is overwhelmed, the command will timeout in 10s and then we will still retry every 5s - but only until the aggregate time is less than 30s (at least that is how I think it would work)